x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexCellIdSource.java

```diff
@@ -145,5 +101,16 @@ private boolean intersects(double minLon, double maxLon, double minLat, double m
             }
             return false;
         }
+
+        public boolean validPoint(double lat, double lon) {
```

Comment:
Should be private

Reply:
Also, this method, and the one in `GeoTileCellIdSource`, could move to `CellIdSource` parent class, which can also have `crossesDateline`. Since two of the three child classes use the same function, perhaps it makes sense. However, if `GeoTileCellIdSource`  stops using `validPoint`, then there is no point in moving this.

Reply:
I like the idea of moving this method to `CellIdSource`, I have a go